gulpfile.js

```diff
@@ -0,0 +1,46 @@
+(function () {
```

Comment:
That's very interesting, and I really like that it'll be more modular and maintainable..  Let me try to break it up into lint, test, and watch tasks.